src/metabase/driver/common/parameters/values.clj

```diff
@@ -114,12 +114,25 @@
                                              params)]
                              (if (= (count params) 1)
                                (first params)
-                               params)))]
+                               params)))
+        nil-value?        (and (seq matching-params)
+                               (every? (fn [param]
+                                         (nil? (:value param)))
+                                       matching-params))]
     (or
```

Comment:
It'd be super nice if we can refactor this into a cond.

It's very hard to reason multi-layers of conditions like this